File: two_pointers/392_is_subsequence.py
def isSubsequence(s: str, t: str) -> bool:
        p1 = 0

        for c in t:
            # если уже вышли за границу
            if p1 == len(s):
                return True
            if c == s[p1]:
                p1 += 1
        
        return p1 == len(s)

File: two_pointers/392_is_subsequence.go
func isSubsequence(s string, t string) bool {
	p1 := 0

	for p2 := range len(t) {
		// если уже вышли за границу
		if p1 == len(s) {
			return true
		}
		if s[p1] == t[p2] {
			p1++
		}
	}

	return p1 == len(s)
}